### PR TITLE
feat: update assigned message #407

### LINKED ISSUE #407: Update Assigned Message
When a bounty hunter is assigned the new message should include the total bounty compensation amount. This should recalculate the bounty based on 1. their `multiplier` and 2. the `base-rate`.

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquibot/issues/379#issuecomment-1593746208_

## New comment template

<code>

<table>
<tbody>
<tr><td>Deadline</td><td>Thu, 15 Jun 2023 10:47 UTC</td></tr>
<tr><td>Registered Wallet</td><td>0x4007CE2083c7F3E18097aeB3A39bb8eC149a341d</td></tr>
<tr><td>Payment Multiplier</td><td>1.00</td></tr>
<tr><td>Multiplier Reason</td><td>Reason</td></tr>
<tr><td>Total Bounty</td><td>100 USD</td></tr>
</tbody></table>
</code>

<h6>Tips:</h6>
<ul>
<li>Use <code>/wallet 0x4FDE...BA18</code> if you want to update your registered payment wallet address @user.</li>
<li>Be sure to open a draft pull request as soon as possible to communicate updates on your progress.</li>
<li>Be sure to provide timely updates to us when requested, or you will be automatically unassigned from the bounty.</li>
<ul>

### PR DESCRIPTION
**Pull request by starlitnightsky**
Resolves #407 

- Test
https://github.com/starlitnightsky/Real-Estate/issues/10#issuecomment-1597911250

### CODE REVIEW

File: src/handlers/comment/handlers/assign.ts [hunk 1]
```diff
@@ -84,6 +86,10 @@ export const assign = async (body: string) => {
   const curDate = new Date();
   const curDateInMillisecs = curDate.getTime();
   const endDate = new Date(curDateInMillisecs + duration * 1000);
+  const deadline_msg = endDate.toUTCString();
+  const reason_msg = issue.state_reason ? issue.state_reason : "Reason";
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/assign.ts:90]: I think you should get the `reason` from the `wallets` table.

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

File: src/handlers/comment/handlers/assign.ts (hunk 1 above)
Comment (starlitnightsky) [reply src/handlers/comment/handlers/assign.ts]: Resolves comment
https://github.com/starlitnightsky/Real-Estate/issues/10#issuecomment-1598049354
Default Reason is null, so there is no reason at first, I think.

File: src/handlers/comment/handlers/assign.ts [hunk 2]
```diff
@@ -101,17 +107,26 @@ export const assign = async (body: string) => {
...
+    }
+    const issueDetailed = bountyInfo(issue);
+    if (!issueDetailed.priceLabel) {
+      bounty_msg = `Permit generation skipped since price label is not set`;
+    } else {
+      bounty_msg = (+issueDetailed.priceLabel!.substring(7, issueDetailed.priceLabel!.length - 4) * multiplier).toString() + " USD";
     }
-    return commit_msg;
+    const reason = await getWalletReason(payload.sender.login);
+    reason_msg = reason ? reason : "";
```
Comment (0xcodercrane) [inline src/handlers/comment/handlers/assign.ts:128]: you can make it simple
```suggestion
    reason_msg = reason ?? "";
```

**Review by 0xcodercrane — APPROVED** [review]

**Review by Draeieg — APPROVED** [review]

File: src/adapters/supabase/helpers/client.ts [hunk 3]
```diff
@@ -271,3 +271,9 @@ export const getWalletMultiplier = async (username: string): Promise<number> =>
   if (data?.multiplier == null) return 1;
   else return data?.multiplier;
 };
+
+export const getWalletReason = async (username: string): Promise<string> => {
```
Comment (pavlovcik) [inline src/adapters/supabase/helpers/client.ts:275]: This should be `getMultiplierReason`

Comment (0xcodercrane) [reply src/adapters/supabase/helpers/client.ts:275]: https://github.com/ubiquity/ubiquibot/commit/1c14efc29cc15820fe900e491112570723ef72de

**Pull request merged**